### PR TITLE
Add r2s example

### PR DESCRIPTION
**Pull request by zxkjack123**
This PR adds a PyNE r2s example and the usage of PyNE r2s example files.
Major changes:

- Add PyNE R2S example files in `pyne/examples`.

- Update r2s documentation in `pyne/docs/userguide/r2s.rst`.

### CODE REVIEW

File: docs/usersguide/r2s.rst [hunk 1]
```diff
@@ -113,6 +113,94 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
 Note that each of these source files must be renamed to "source.h5m" for this purpose.
 By using these sources for photon transport, the shutdown dose rate can be obtained. Tally results will have to be normalized by the total photon source intentity. This information is found in the "total_photon_source_intensites.txt" file printed out by r2s.py step2.
 
+***************
+PyNE R2S example
+***************
```
Comment (scopatz) [inline docs/usersguide/r2s.rst]: Title lines need to cover the entire title

File: docs/usersguide/r2s.rst [hunk 2]
```diff
@@ -113,6 +113,94 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+    :align: center
+
+    **Figure 1:** *Trelis model for the example.*
+
+
+Prepare material library, build material library. The material lib is then generated in file "example_material_lib.h5".
+
+.. code-block:: bash
+
+   >> python make_example_material.py
```
Comment (scopatz) [inline docs/usersguide/r2s.rst]: It is more traditional to use `$` to indicate the prompt, rather than `>>`

**Comment by scopatz** [comment]
This PR seems to have many binary files (lib, h5m, etc) commited.  Please remove these AND remove them from the history so that they are not included in the repository. Thanks for putting this in @zxkjack123!

**Comment by zxkjack123** [comment]
@scopatz These binary files are necessary part of the example. Is there any suitable place to put these files in PyNE project?

**Comment by scopatz** [comment]
@zxkjack123, 

1. can you not generate the examples?
2. we have been sticking binaries in the `pyne/data` repo

File: docs/usersguide/r2s.rst [hunk 3]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+PyNE R2S example
+****************
+
+Using a simple geometry as a example, here is how we perform R2S calculation.
+The example geometry composed of four cubes of size 10x10x10 cm3. Cubes (in green) at bottom left and up right are water. Bottom right cube (in red) is made of steel, and up left cube (in white) is void.
+There is a isotropic single energy (14 MeV) neutron point source in the middle of the bottom right cube.
+The cubes are irradiated by the source with a single 3.5 day pulse, the neutron intensity is 1e10 n/s.
+The following content demonstrates the process of caculate the shutdown dose rate of these cubes at the time of 1 hour after shutdown.
+Example files could be found in `r2s_example <https://github.com/pyne/data/blob/master/r2s_example.tar.gz>`_.
+>`_.
```
Comment (gonuke) [inline docs/usersguide/r2s.rst]: Extra line here by accident with left over formatting

File: docs/usersguide/r2s.rst [hunk 4]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+
+
+.. figure:: r2s_example_geometry.png
+    :align: center
+
+    **Figure 1:** *Geometry of R2S example (X-Y cross section).*
+
+Build the model using trelis as following figure. Set material group and export to "geom_without_material.h5m".
```
Comment (gonuke) [inline docs/usersguide/r2s.rst]: use back ticks here to get better formatting of the filename

File: docs/usersguide/r2s.rst [hunk 5]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+
+Combine the geometry file and the material library, using  the following command:
+
+.. code-block:: bash
+
+   $ cp geom_without_material.h5m geom.h5m
+
+   $ uwuw_preproc geom.h5m -v -l example_material_lib.h5
+
+Prepare input file, define source, tally and other data cards. Example input file could be seen in r2s_example/neutron_transport/input.
```
Comment (gonuke) [inline docs/usersguide/r2s.rst:156]: Link to this file?

File: docs/usersguide/r2s.rst [hunk 6]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+
+.. code-block:: bash
+
+   $ cp r2s_run
+
+   $ ln -sf ../neutron_transport/meshtal .
+
+   $ r2s.py setup
+
+Modify important parameters in the 'alara_params.txt' and 'config.ini' according to the problem. Examples could see 'alara_params_example.txt' and 'config_example.ini' in 'r2s_example/r2s_run'.
```
Comment (gonuke) [inline docs/usersguide/r2s.rst]: "could be seen"

Comment (gonuke) [reply docs/usersguide/r2s.rst]: link to the directory?

File: docs/usersguide/r2s.rst [hunk 7]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+Prepare alara nuclide library, copy preinstalled data library from ALARA repository. Example nuclide library could be seen in 'r2s_example/r2s_run/data'
```
Comment (gonuke) [inline docs/usersguide/r2s.rst]: link to this?

File: docs/usersguide/r2s.rst [hunk 8]
```diff
@@ -113,6 +113,96 @@ that can utilize these mesh-based sources is found in the PyNE user's guide entr
...
+
+   $ r2s.py step1
+
+Perform R2S step2. Several photon source file and a e_bounds file will be generated in this step.
+
+.. code-block:: bash
+
+   $ r2s.py step2
+
+Perform Photon transport calculation. Example input file could be seen in r2s_example/photon_transport/input.
```
Comment (gonuke) [inline docs/usersguide/r2s.rst:195]: link to thisi

**Review by gonuke — CHANGES_REQUESTED** [review]
Add a few more links

File: docs/usersguide/r2s.rst (hunk 5 above)
Comment (zxkjack123) [reply docs/usersguide/r2s.rst:156]: All the example files are in a `r2s_example.tar.gz`, there isn't a link for this `input` file.

**Comment by zxkjack123** [comment]
@gonuke Some of the format and links are modified according to your comment. However, I can't add some links as you requested. Because these files are all in the `r2s_example. tar.gz`, there isn't a link for a single file.

**Review by gonuke — APPROVED** [review]

**Comment by gonuke** [comment]
Thanks @zxkjack123

**Comment by gonuke** [comment]
Someone needs to rebuild the webpage - there may be instructions somewhere??? and/or a docker image to help?

**Comment by scopatz** [comment]
We should probably cut a release, too. That would update the website.

**Pull request merged**